pubtools/_pulp/hooks.py

```diff
@@ -14,4 +14,23 @@ def task_pulp_flush():
     Pulp content may have recently changed."""
 
 
+@hookspec
+def pulp_item_push_finished(
+    pulp_units=None, push_item=None
+):  # pylint: disable=unused-argument
+
+    """Invoked during push tasks after each item has been processed fully.
+
+    By the time this hook is invoked, the referenced item and unit is expected
+    to be fully uploaded into Pulp and published onto the CDN.
+    Args:
+        pulp_units (list[:class:`~pubtools.pulplib.Unit`])
+            A list of zero or more Pulp unit(s) created/updated for this item.
```

Comment:
There is currently a mismatch between the docs and what you implemented.

Docs say that it's a list of zero or more, and doesn't say anything about it possibly being None.  This implies that if there's no units, it'll give you an empty list.

What you've implemented instead is that pulp_units will be None if there were no units.

Could you please make the docs and implementation match?  Probably by yielding an empty list rather than None in the 'no units' case.